### DISCUSSION TITLE
Need help in understand the project structure and architecture.

### DISCUSSION THREAD

**Discussion by alanbosco003** [zed-industries/zed#12347] [Feature Requests]
I'm very new to Rust and this repository. I wish to contribute to this repository. But I don't know where to start. It would be very helpful if there were any pages I could refer to understand the patterns and architecture used. currently, I don't have any Idea where to start.

**Comment by bajrangCoder** [comment]
Zed uses a Rust crates member-like(or cargo workspaces) approach, breaking the project into multiple crates, which are stored in the `crates` directory. And `crates/zed` is the main entry point .

For more information on the workspaces approach in Rust, refer to the [Rust documentation on Workspaces.](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html#cargo-workspaces)

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
